libvast/vast/concept/parseable/vast/type.hpp

```diff
@@ -118,15 +119,17 @@ struct type_parser : parser<type_parser> {
       return vector_type{std::move(value_type)}.attributes(std::move(attrs));
     };
     auto vector_type_parser
-      = ("vector" >> ws >> '<' >> ws >> type_type >> ws >> '>') ->* to_vector
+      = ("vector" >> ws >> '<' >> ws >> vast::ref(type_type) >> ws >> '>')
```

Comment:
Could `using vast::ref` help here to bring `ref` in scope and make the grammar definition as terse as possible?

Reply:
Please note that we are already in namespace `vast`. I'm not quite sure why `std::ref` becomes a candidate function in the first place, but without qualifier is gets precedence.

Reply:
I've added a second overload so that there's always a more specialized `vast::ref` than there's a `std::ref`.